criu/pie/restorer.c

```diff
@@ -1243,6 +1244,73 @@ static inline int restore_child_subreaper(int child_subreaper)
 		return 0;
 }
 
+static int fd_poll(int inotify_fd)
+{
+	struct pollfd pfd = {inotify_fd, POLLIN, 0};
+#ifdef SYS_poll
+	return sys_poll(&pfd, 1, 0);
+#else
+	struct timespec tmo = {0, 0};
+
+	return sys_ppoll(&pfd, 1, &tmo, NULL, sizeof(sigset_t));
+#endif
+}
+
+/*
+ * note: Actually kernel may want even more space for one event (see
+ * round_event_name_len), so using buffer of EVENT_BUFF_SIZE size may fail.
+ * To be on the safe side - take a bigger buffer, and these also allows to
+ * read more events in one syscall.
+ */
+#define EVENT_BUFF_SIZE ((sizeof(struct inotify_event) + PATH_MAX))
+
+/*
+ * Read all available events from inotify queue
+ */
+static int cleanup_inotify_events(int inotify_fd)
+{
+	char buf[EVENT_BUFF_SIZE * 8];
+	int ret;
+
+	while (1) {
+		ret = fd_poll(inotify_fd);
+		if (ret < 0) {
+			pr_err("Failed to poll from inotify fd\n");
+			return -1;
+		} else if (ret == 0) {
+			break;
+		}
+
+		ret = sys_read(inotify_fd, buf, sizeof(buf));
+		if (ret < 0) {
+			pr_err("Failed to read inotify events\n");
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+/*
+ * When we restore inotifies we can open and close files we create a watch
+ * for. So wee need to cleanup these auxiliary events which we've generated.
```

Comment:
Pls add in a comment that we don't restore inotify events, so we can just clean all events from a queue.

Reply:
done.